src/rt-app.c

```diff
@@ -849,6 +849,105 @@ static void set_thread_affinity(thread_data_t *data, cpuset_data_t *cpu_data)
 	}
 }
 
+#if HAVE_LIBNUMA
+static int create_numaset_str(numaset_data_t *numa_data)
+{
+	unsigned int i;
+	unsigned int idx = 0;
+
+	int node_count = numa_bitmask_weight(numa_data->numaset);
+
+	/*
+	 * Assume we have less than 99 NUMA nodes
+	 * in the system.
+	 * Each node would take up to 2 + 2
+	 * bytes (2 for the number and 2 for the comma and space). 2 bytes
+	 * for beginning bracket + space and 2 bytes for end bracket and space
+	 * and finally null-terminator.
+	 */
+	unsigned int size_needed = node_count * 4 + 2 + 2 + 1;
+
+	numa_data->numaset_str = malloc(size_needed);
+	if (!numa_data->numaset_str) {
+		log_error("Failed to set numaset string");
+		return -1;
+	}
+
+	strcpy(numa_data->numaset_str, "[ ");
+	idx += 2;
+
+	for (i = 0; i < 100 && node_count; ++i) {
+		unsigned int n;
+
+		if (numa_bitmask_isbitset(numa_data->numaset, i)) {
+			--node_count;
+			if (size_needed <= (idx + 1)) {
+				log_error("Not enough memory for array");
+				exit(EXIT_FAILURE);
+			}
+			n = snprintf(&numa_data->numaset_str[idx],
+						size_needed - idx - 1, "%u", i);
```

Comment:
Is the s in size_needed supposed to be lined up with the n from numaset? Or is this strictly tab based alignment?